src/animations/constructor.js

```diff
@@ -0,0 +1,64 @@
+// Require react with addons to get access to React Transition Groups
+var React = require("react/addons");
+
+//Require arrival to know when elements and their children have transitionended.
+var Arrival = require('Arrival');
+
+// Creates an object that holds the React Transition Group Component
+var ReactTransitionGroup = React.addons.TransitionGroup;
+
+// Constructor function that can be invoked to create custom animation components.
+var makeAnimation = function(displayName, transitions) {
+  return React.createClass({
+    componentWillEnter: function(done) {
+      // This is called at the same time as componentDidMount() for components added to an existing TransitionGroup. It will block other animations from occurring until callback is called. It will not be called on the initial render of a TransitionGroup.
+      this.el = this.getDOMNode();
```

Comment:
Recommend refactoring out lines 15 and 16 into a separate method, as it gets used here and in componentWillLeave.